app/components/post_draft/quick_actions/camera_quick_action/photo_attacment.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+import {Text} from 'react-native';
+
+import CompassIcon from '@app/components/compass_icon';
+import TouchableWithFeedback from '@app/components/touchable_with_feedback';
+import {ICON_SIZE} from '@app/constants/post_draft';
+import {useTheme} from '@app/context/theme';
+import {dismissBottomSheet} from '@app/screens/navigation';
+import FilePickerUtil from '@app/utils/file/file_picker';
+import {preventDoubleTap} from '@app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@app/utils/theme';
+
+import type {QuickActionAttachmentProps} from '@typings/components/post_draft_quick_action';
+import type {CameraOptions} from 'react-native-image-picker';
+
+const getStyle = makeStyleSheetFromTheme((theme: Theme) => ({
+    title: {
+        color: theme.centerChannelColor,
+        fontSize: 16,
+        fontWeight: 400,
+        lineHeight: 24,
+    },
+    container: {
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+        gap: 16,
+        paddingVertical: 12,
+    },
+}));
+
+const PhotoAttachment: React.FC<QuickActionAttachmentProps> = ({
+    disabled,
+    onUploadFiles,
+    maxFilesReached,
+    testID,
+}) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const style = getStyle(theme);
+
+    const handleButtonPress = useCallback((options: CameraOptions) => {
+        const picker = new FilePickerUtil(intl,
+            onUploadFiles);
+
+        picker.attachFileFromCamera(options);
+    }, [intl, onUploadFiles]);
+
+    const onPhoto = async () => {
+        const options: CameraOptions = {
+            quality: 0.8,
+            mediaType: 'photo',
+            saveToPhotos: true,
+        };
+
+        await dismissBottomSheet();
+        handleButtonPress(options);
+    };
+
+    const onPress = useCallback(preventDoubleTap(onPhoto, 500), []);
```

Comment:
These three functions can be simplified into one. Something like this:
```suggestion
    const onPress = useCallback(preventDoubleTap(async () => {
        await dismissBottomSheet();
        
        const picker = new FilePickerUtil(intl,
            onUploadFiles);  
        const options: CameraOptions = {
            quality: 0.8,
            mediaType: 'photo',
            saveToPhotos: true,
        };
        picker.attachFileFromCamera(options);
    }, 500, [intl, onUploadFiles]);
```

Reply:
Sure, Done..